Noishadha Chatterjee/cst438-code-review-noishadha/src/Main.java

```diff
@@ -0,0 +1,202 @@
+                    break;
+                default:
+                    System.out.println("Invalid choice. Please try again.");
+            }
+        }
+    }
+
+    //Load Data from File
+    private static void loadData() {
+        readUser("src/user-info.txt");
+        readPost("src/post-info.txt");
+        System.out.println("Input data loaded successfully.");
+    }
+
+    //Read User Info Data from File
+    private static void readUser(String filename) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(";");
+                String username = parts[0];
+                String displayName = parts[1];
+                String state = parts[2];
+                Set<String> friends = new HashSet<>(Arrays.asList(parts[3].replaceAll("[\\[\\]]", "").split(", ")));
+                users.add(new User(username, displayName, state, friends));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    //Read Post Info from File
+    private static void readPost(String filename) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(";");
+                String postId = parts[0];
+                String userId = parts[1];
+                String visibility = parts[2];
+                posts.add(new Post(postId, userId, visibility));
+            }
+        } catch (IOException e) // File Reader Exceptions Handler .
+        {
+            e.printStackTrace();
+        }
+    }
+
+    // Method to Check Visibility of a Post relative to the User
+    private static boolean canUserViewPost(String username, String postId) {
+        for (Post post : posts) {
+            if (post.postId.equals(postId))
+            {
+                if (post.visibility.equals("public"))
+                {
+                    return true;
+                }
+                else if (post.visibility.equals("friend"))
+                {
+                    User user = getUserByUsername(username);
+                    return user != null && user.friends.contains(post.userId);
+                }
+            }
+        }
+        return false;
+    }
+
+    // Method to retrieve accessible posts from User .
+    private static List<Post> getAccessiblePosts(String username) {
+        List<Post> accessiblePosts = new ArrayList<>();
+        for (Post post : posts) {
+            if (post.userId.equals(username) || canUserViewPost(username, post.postId)) {
+                accessiblePosts.add(post);
+            }
+        }
+        return accessiblePosts;
+    }
+
+
+    // Method to get users by location
+    private static List<User> getUsersByLocation(String state) {
```

Comment:
could just move this into the searchUsersByLocation() function since that is the only place it is used